integration/docker/volume_test.go

```diff
@@ -90,4 +93,50 @@ var _ = Describe("docker volume", func() {
 			Expect(stdout).NotTo(ContainSubstring(testFile))
 		})
 	})
+
+	Context("creating a text file under /dev", func() {
+		It("should be passed the content to the container", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create files under /dev")
+			}
+			fileName := "/dev/foo"
+			textContent := "hello"
+			err = ioutil.WriteFile(fileName, []byte(textContent), 0644)
+			Expect(err).ToNot(HaveOccurred())
+			defer os.Remove(fileName)
+
+			args = []string{"--name", id, "-v", fileName + ":" + fileName, Image, "cat", fileName}
+			stdout, _, exitCode = dockerRun(args...)
+			Expect(exitCode).To(Equal(0))
+			Expect(stdout).To(ContainSubstring(textContent))
+
+			Expect(RemoveDockerContainer(id)).To(BeTrue())
+			Expect(ExistDockerContainer(id)).NotTo(BeTrue())
+		})
+	})
+
+	Context("passing a block device", func() {
+		It("should be mounted", func() {
+			if os.Getuid() != 0 {
+				Skip("only root user can create loop devices")
+			}
+
+			diskFile, loopFile, err = createLoopDevice()
+			Expect(err).ToNot(HaveOccurred())
+
+			args = []string{"--name", id, "--cap-add=SYS_ADMIN", "-v", loopFile + ":" + loopFile, DebianImage, "mount", loopFile, "/mnt"}
```

Comment:
to have `mount` working you have to format the loop device before mounting, so you have to do something like this

```go
loopFileP1 = fmt.Sprintf("%sp1", loopFile)
mkfsCmd := tests.NewCommand("mkfs.ext4", loopFileP1)
if _, stderr, exitCode := mkfsCmd.Run(); exitCode != 0 {
	return "", "", fmt.Errorf("%s", stderr)
}
args = []string{"--name", id, "--cap-add=SYS_ADMIN", "-v", loopFileP1 + ":" + loopFileP1, DebianImage, "mount", loopFileP1, "/mnt"}
```

Reply:
@devimc thanks for the feedback

Reply:
updated example code